frontend/openchat-agent/src/services/openchatAgent.ts

```diff
@@ -276,32 +278,50 @@ export class OpenChatAgent extends EventTarget {
     ) {
         super();
         this._logger = config.logger;
+        this._agent = this.createHttpAgent();
         this.db = initDb(this.principal);
-        this._onlineClient = OnlineClient.create(identity, config);
-        this._userIndexClient = new UserIndexClient(identity, config);
-        this._groupIndexClient = GroupIndexClient.create(identity, config);
-        this._notificationClient = NotificationsClient.create(identity, config);
-        this._proposalsBotClient = ProposalsBotClient.create(identity, config);
-        this._marketMakerClient = MarketMakerClient.create(identity, config);
-        this._registryClient = RegistryClient.create(identity, config);
-        this._icpcoinsClient = ICPCoinsClient.create(identity, config);
-        this.translationsClient = new TranslationsClient(identity, config);
-        this._signInWithEmailClient = SignInWithEmailClient.create(identity, config);
-        this._signInWithEthereumClient = SignInWithEthereumClient.create(identity, config);
-        this._signInWithSolanaClient = SignInWithSolanaClient.create(identity, config);
+        this._onlineClient = new OnlineClient(identity, this._agent, config);
+        this._userIndexClient = new UserIndexClient(identity, this._agent, config);
+        this._groupIndexClient = new GroupIndexClient(identity, this._agent, config);
+        this._notificationClient = new NotificationsClient(identity, this._agent, config);
+        this._proposalsBotClient = new ProposalsBotClient(identity, this._agent, config);
+        this._marketMakerClient = new MarketMakerClient(identity, this._agent, config);
+        this._registryClient = new RegistryClient(identity, this._agent, config);
+        this._dataClient = new DataClient(identity, this._agent, config);
+        this._icpcoinsClient = new ICPCoinsClient(identity, this._agent);
+        this.translationsClient = new TranslationsClient(identity, this._agent, config);
+        this._signInWithEmailClient = new SignInWithEmailClient(identity, this._agent, config);
+        this._signInWithEthereumClient = new SignInWithEthereumClient(
+            identity,
+            this._agent,
+            config,
+        );
+        this._signInWithSolanaClient = new SignInWithSolanaClient(identity, this._agent, config);
         this._localUserIndexClients = {};
         this._ledgerClients = {};
         this._ledgerIndexClients = {};
         this._groupClients = {};
         this._communityClients = {};
-        this._dexesAgent = new DexesAgent(config);
+        this._dexesAgent = new DexesAgent(this._agent);
         this._groupInvite = config.groupInvite;
     }
 
     private get principal(): Principal {
         return this.identity.getPrincipal();
     }
 
+    private createHttpAgent(): HttpAgent {
```

Comment:
might be nice if this `createHttpAgent` were just a shareable function rather than a method since it is also used (or something very similar in the identity agent and we'd want it as difficult as possible to get wrong (the whole fetchRootKey thing has been wrong in past).

Reply:
Done